.github/workflows/docker-and-k8s.yml

```diff
@@ -87,93 +87,67 @@ jobs:
-#          mv ~/docker-image-build-cache ~/docker-image-build-cache-in
-#          # If cache does not exist, then create a new empty cache directory,
-#          # where a new build cache will be stored.
-#          mkdir -p ~/docker-image-build-cache
-#
-#      - name: Push image using the master branch commit SHA.
-#        if: ${{ github.ref == 'refs/heads/master' }}
-#        run: |
-#          # NOTE: --remove-image-name-prefix is needed in case Docker hub usename is not scalyr
-#          python3 build_package_new.py "${{ inputs.image-type }}-${{ matrix.image-distro-name }}" \
-#            --registry "${{ secrets.DOCKER_HUB_USERNAME }}" \
-#            --remove-image-name-prefix \
-#            --push \
-#            --tag "${{ github.sha }}${{ matrix.image_tag_suffix_full }}" \
-#            --cache-from-dir ~/docker-image-build-cache-in \
-#            --cache-to-dir ~/docker-image-build-cache
-#
-#      - name: Push image using the tag.
-#        if: ${{ github.ref_type == 'tag' }}
-#        run: |
-#          tag_options=$(python3 scripts/cicd/verify_and_get_image_tag_to_publish.py "${{ github.ref_name }})" --sufix "${{ matrix.image_tag_suffix }}"
-#
-#          python3 build_package_new.py ${{ inputs.image-type }} \
-#            --registry "${{ secrets.DOCKER_HUB_USERNAME }}" \
-#            --remove-image-name-prefix \
-#            --push \
-#            --cache-from-dir ~/docker-image-build-cache-in \
-#            --cache-to-dir ~/docker-image-build-cache \
-#            ${tag_options}
-#
-#          # Check the cache directory size post build
-#          du -hs ~/docker-image-build-cache
-#          ls -lah ~/docker-image-build-cache
+  publish-temp-release:
+    needs: [test]
+    name: Publish Docker Image
+    if: ${{ inputs.publish == true && (github.ref == 'refs/heads/improve-caching' || github.ref_type == 'tag') }}
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        image-type: [ "docker-json", "docker-syslog", "docker-api", "k8s" ]
+        image-distro-name: [ "buster", "alpine" ]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        id: qemu
+        uses: docker/setup-qemu-action@v1
+        with:
+          image: tonistiigi/binfmt:qemu-v6.1.0
+          platforms: all
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          driver-opts: network=host
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+
+      - name: Push image using the master branch commit SHA.
+        if: ${{ github.ref == 'refs/heads/master' }}
+        run: |
+
+          # Push result docker image with a special tag "_temp_release".
```

Comment:
Would be good to add a note here that this is needed because this step doesn't just push the image, but it also builds it (which may not be immediately obvious since the step name just stay push, maybe we should update it to say Build and push docker image using master branch commit sha).

Reply:
Yeah, it also builds it, I'll add that. Another notion is that it does not actually push using master branch commit sha. The main purpose of this job is just to build and push to the docker hub with the `_temp_release` tag if we are in the master branch or some tag. The following job has to run a small script file `scripts/cicd/publish-docker-images-tags.py` that has to re-tag previously pushed `_temp_release` images to normal tags.

Another note: I have also added a concurrency option to these jobs, to avoid race conditions on multiple parallel pushes.

Reply:
Ack, thanks for clarifying.